Update SysId Encoder docs (#1506)
wpilibsuite/frc-docs: source/docs/software/pathplanning/robot-characterization/configuring-project.rst

Encoder Selection
^^^^^^^^^^^^^^^^^
The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`).

.. image:: images/encoder-selection.png
   :alt: Encoder selection section.

Encoder Types
"""""""""""""

> - **General Types** (consistent across all motor controller selections): ``roboRIO`` corresponds to any encoders that are plugged into the rio, ``CANCoder`` corresponds to the CTRE CANCoder.

Reviewer: ```suggestion
- **General Types** (consistent across all motor controller selections): ``roboRIO`` corresponds to any encoders that are plugged into the roboRIO, ``CANCoder`` corresponds to the CTRE CANCoder.
```

> - **TalonSRX** or **TalonFX**: ``Built-in`` corresponds to any quadrature encoder plugged into the encoder port, ``Tachometer`` corresponds to the CTRE Tachometer plugged into the encoder port.

Reviewer: The built-in encoder for the Talon FX isn't actually a quadrature encoder.
```cpp
  if (encoderType == "Built-in") {
    if (starts_with(controllerName, "Talon")) {
      FeedbackDevice feedbackDevice;
      if (controllerName == "TalonSRX") {
        feedbackDevice = FeedbackDevice::QuadEncoder;
      } else {
        feedbackDevice = FeedbackDevice::IntegratedSensor;
      }
```

Encoder Selection
^^^^^^^^^^^^^^^^^
The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`).

.. image:: images/encoder-selection.png
   :alt: Encoder selection section.

Encoder Types

> """""""""""""

Reviewer 2: ```suggestion
~~~~~~~~~~~~~
```

Reviewer 2: We should not have more than 4 layers of depth, it is bad flow.

Encoder Selection
^^^^^^^^^^^^^^^^^

> The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`). However, there are two types that are consistent across all motor controller selections: ``roboRIO`` which corresponds to any encoders that are plugged into the rio, ``CANCoder`` which corresponds to the CTRE CANCoder. If you are using a TalonSRX or TalonFX for your motor controller, the ``Built-in`` corresponds to any quadrature encoder plugged into the encoder port and the ``Tachometer`` option corresponds to the CTRE Tachometer plugged into the port. If you're using a Spark MAX, the ``Encoder Port`` refers to an encoder plugged into the `encoder port <https://docs.revrobotics.com/sparkmax/feature-description/encoder-port>`__ and the ``Data Port`` refers to an encoder plugged into the `data port <https://docs.revrobotics.com/sparkmax/feature-description/data-port>`__. For the Venom, ``Built-in`` refers to an encoder plugged into the Venom's encoder port.

Reviewer 2: I think the latter part of this paragraph can get broken up into a list

> The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`), however there are two types that are consistent across all motor controller selections: ``roboRIO`` which corresponds to any encoders that are plugged into the rio, ``CANCoder`` which corresponds to the CTRE CANCoder. If you are using a TalonSRX or TalonFX for your motor controller, the ``Built-In`` corresponds to any quadrature encoder plugged into the encoder port and the ``Tachometer`` option correspondsor to the CTRE Tachometer plugged into the port. If you're using a SparkMAX, the ``Encoder Port`` refers to an encoder plugged into the :ref:`encoder port <https://docs.revrobotics.com/sparkmax/feature-description/encoder-port>` and the ``Data Port`` refers to an encoder plugged into the :ref:`data port <https://docs.revrobotics.com/sparkmax/feature-description/data-port>`. For the Venom, `Built-in` refers to an encoder plugged into the Venom's encoder port.

Reviewer: ```suggestion
The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`). However, there are two types that are consistent across all motor controller selections: ``roboRIO`` which corresponds to any encoders that are plugged into the rio, ``CANCoder`` which corresponds to the CTRE CANCoder. If you are using a TalonSRX or TalonFX for your motor controller, the ``Built-In`` corresponds to any quadrature encoder plugged into the encoder port and the ``Tachometer`` option corresponds to the CTRE Tachometer plugged into the port. If you're using a Spark MAX, the ``Encoder Port`` refers to an encoder plugged into the :ref:`encoder port <https://docs.revrobotics.com/sparkmax/feature-description/encoder-port>` and the ``Data Port`` refers to an encoder plugged into the :ref:`data port <https://docs.revrobotics.com/sparkmax/feature-description/data-port>`. For the Venom, `Built-in` refers to an encoder plugged into the Venom's encoder port.
```
